src/main/java/org/elasticsearch/transport/couchbase/capi/CouchbaseCAPITransportImpl.java

```diff
@@ -128,7 +152,7 @@ public CouchbaseCAPITransportImpl(Settings settings, RestController restControll
         this.parentSelector.configure(settings);
         logger.info("Couchbase transport is using parent selector: {}", parentSelector.getClass().getCanonicalName());
 
-        Class<? extends KeyFilter> keyFilterClass = settings.<KeyFilter>getAsClass("couchbase.keyFilter", DefaultKeyFilter.class);
+        Class<? extends KeyFilter> keyFilterClass = DefaultKeyFilter.class;
```

Comment:
@Branor See the comment above.

If that change is valid, then this line would become:

    Class<? extends ParentSelector> parentSelectorClass = this.getAsClass(settings.get("couchbase.parentSelector"), DefaultParentSelector.class);